usr/sbin/p11sak/p11sak.c

```diff
@@ -912,6 +927,55 @@ static void set_bool_attr_from_string(CK_ATTRIBUTE *attr, char attr_char)
     }
 }
 
+/**
+ * @brief Set the custom attrs from string object
+ * 
+ * @param attr The attribute array
+ * @param attr_char The string from the user
+ * @param count Index of current attribute in the attribute array
+ */
+static CK_RV set_custom_attrs_from_string(char *attr_char, CK_ATTRIBUTE *attr, CK_ULONG *count) 
+{
+    CK_RV rc = CKR_OK;
+    int i = 0, j = 0;
+    char substr[10];
+
+    // iterate over the single custom attributes
+    while (i < (int)strlen(attr_char))
+    {
+        int radix = 16;
+        unsigned long id = 0;
+
+        // parse attribute id from string
+        for (j = 0; attr_char[i] != '='; i++) {
+            if (j == 1 && attr_char[i] != 'x' && attr_char[i] != 'X') 
+                radix = 10;
+            substr[j] = attr_char[i];
+            j++;  
+        }
+        id = strtoul(substr, NULL, radix);
+        i++;
+
+        // set attribute in array
+        attr[*count].type = id;
+        if (strncasecmp((attr_char + i), "false", 5) == 0) {
```

Comment:
what if I use 'falseblah' as value? You would simply ignore blah and treat ist as false. You would probably detect an error when continuing to scan the string though....

Reply:
I think that parser needs some rework.  Why not go the usual parser way of tokenization along `,`, then parsing each token into its key and value part (splitting along the `=`), and then analyzing these parts.

Reply:
Done